src/engine.rs

```diff
@@ -883,31 +866,7 @@ impl Reedline {
 
     fn event_gen(&self) -> io::Result<ReedlineEvent> {
         let event = read()?;
-        let editor_event = match event {
-            Event::Key(KeyEvent { code, modifiers }) => match (modifiers, code) {
-                (KeyModifiers::NONE, KeyCode::Tab) => ReedlineEvent::HandleTab,
-                (KeyModifiers::CONTROL, KeyCode::Char('d')) => ReedlineEvent::CtrlD,
-                (KeyModifiers::CONTROL, KeyCode::Char('c')) => ReedlineEvent::CtrlC,
-                (KeyModifiers::CONTROL, KeyCode::Char('l')) => ReedlineEvent::ClearScreen,
-                (KeyModifiers::NONE, KeyCode::Char(c))
-                | (KeyModifiers::SHIFT, KeyCode::Char(c)) => {
-                    ReedlineEvent::EditInsert(EditCommand::InsertChar(c))
-                }
-                (KeyModifiers::NONE, KeyCode::Enter) => ReedlineEvent::Enter,
-                _ => {
-                    if let Some(binding) = self.find_keybinding(modifiers, code) {
-                        ReedlineEvent::Edit(binding)
-                    } else {
-                        ReedlineEvent::Edit(vec![])
-                    }
-                }
-            },
-
-            Event::Mouse(_) => ReedlineEvent::Mouse,
-            Event::Resize(width, height) => ReedlineEvent::Resize(width, height),
-        };
-
-        Ok(editor_event)
```

Comment:
And finally, we achieve complete segregation of even parsing into a separate struct.